modules/node_groups/outputs.tf

```diff
@@ -0,0 +1,25 @@
+output "node_groups" {
+  description = "Outputs from EKS node groups. Map of maps, keyed by var.node_groups keys"
+  value = {
+    for k, v in aws_eks_node_group.workers : k => {
+      # values set/generated by terraform-aws-eks
+      iam_role_arn = v.node_role_arn
+      name         = v.node_group_name
+      # output from AWS
+      arn       = v.arn
+      id        = v.id
+      status    = v.status
+      resources = v.resources
+    }
+  }
+}
+
+output "aws_auth_roles" {
+  description = "Roles for use in aws_auth ConfigMap"
+  value = [
```

Comment:
Without some sort of ordering function you're going to hit consistency problems when attempting to convert an unordered map into a list. Is there any reason _not_ to use a map instead of this?

Reply:
Terraform processes the keys in a map in lexical ordering. There aren't consistency problems here.

This is the list of `vars` maps given to the `worker-role.tpl` template file in `aws_auth.tf`. The API is set by that code instead of needing a conversion elsewhere.

I did experiment with generating the aws-auth snippet directly in this module but it felt wrong. I didn't like the use of string concatenation. I think improvements could be made to how the aws-auth ConfigMap is generated but that's out of scope for this PR.